honeycombio/opentelemetry-go-instrumentation
# update smoke tests (e2e tests)

## Which problem is this PR solving?

- need for clearer testing to increase confidence in instrumentation

## Short description of the changes

- add helper function `assert_not_empty` that compares to an empty string. right now it logs the value either way
- use variable `LIBRARY_NAME` in tests to reduce copy-pasta errors and such
- add tests for other checks like expected http attributes

Note I added TODOs for 2 big things: 1) gorilla/mux spans are actually using the net/http probe, and 2) http span names do not match latest conventions.

## How to verify that this has the expected result

e2e tests pass with expected results. Running locally with `make fixture-nethttp` and the others has expected results, and also creates a clean `traces.json` with expected output.

To see an example of a fail locally, navigate to `test/e2e/gin`, change the local `traces.json` and run `bats verify.bats`.

![e2e-fail-gin](https://user-images.githubusercontent.com/29520003/235775910-eee8f08e-b435-4b46-8a6a-6650833d9e5b.png)
